Add support for Win10 defaults and non-Anonymous access

tinkerfairy opened the pull request:
I've constructed a fully qualified path to the share, to get rid of the nt_status_duplicate_name errors I was getting a lot of.

I've also added optional credentials, for machines that disallow anonymous sessions.

cldrn commented:
Did the library arguments smbuser and smbpass not overwrite the authentication headers?

Thanks for the patch!

cldrn commented:
I believe the arguments for user and password are not necessary as the SMB library arguments should be used to configure the script. Can you confirm these don't work for you? (smbuser and smbpass)

Thanks.

aaomidi commented:
Unfortunately this seems to fail on specifying the domain. And I think the issue with that is the specific smb lua library.

cldrn commented:
Yes, people reported me that still doesn't work on Windows 10 despite this patch.

tinkerfairy commented:
I have tested the behavior of the library-supplied "smbusername=,smbdomain=,smbpassword=" and they work the way I want. I have removed the custom arguments and updated the help text.
 (smbdomain= doesn't seem to  be fully functional but that's outside the scope of this patch)

This patch isn't sufficient to test Win10 machines by itself. A modification needs to be made to work around an apparent bug in smb library code. Short term kludge is to set sp_nego = true on line 1319 of nselib/smb.lua (this probably breaks other things). This issue has been reported to dev@

tinkerfairy commented:
Have you had a chance to play with this diff (and the required hack to nselib/smb.lua)?

cldrn commented:
Hey @tinkerfairy , 

Thanks for all your feedback! I've been able to confirm your workaround fixes the issue also for Win 8.1. Win10 also connects to IPC$ fine with this. I'll start looking what's wrong with sp_nego now.

cldrn commented:
Ok, I have fixed the issue and added a check to detect the error code patch systems return! I'll post it later tonight.

Thanks !

oorozcoo commented:
Hey @cldrn 
I think that you are printing some debugs.
|_smb-vuln-ms17-010: 1330795077

tinkerfairy commented:
@cldrn I've tested that your commits to main fix all the issues I opened this PR for. Thanks for the great work!
